spec/defines/ppa_spec.rb

```diff
@@ -43,6 +43,68 @@ def ppa_exec_params(user, repo, distro = 'trusty', environment = [])
     }
   end
 
+  [
+    'ppa:foo/bar',
+    'ppa:foo/bar1.0',
+    'ppa:foo10/bar10',
+    'ppa:foo-/bar_',
+  ].each do |value|
+    describe 'valid resource names' do
+      let :facts do
+        {
+          os: {
+            family: 'Debian',
+            name: 'Ubuntu',
+            release: {
+              major: '18',
+              full: '18.04',
+            },
+            distro: {
+              codename: 'trusty',
+              id: 'Ubuntu',
+            },
+          },
+        }
+      end
+
+      let(:title) { value }
+
+      it { is_expected.not_to raise_error }
```

Comment:
For the happy path, we could also check if the `exec` is being called with the resource title that we expect.

There are a few examples in this spec that use the `contains_exec` method.